geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -53,10 +70,68 @@ class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
         NotFound(reason = s"Unknown ID $id")
       } else {
         contentType = "application/octet-stream"
-        response.setHeader("Content-Disposition", "attachment; filename=" + filename)
+        response.setHeader("Content-Disposition", "attachment;filename=" + filename)
 
         Ok(returnBytes)
       }
     }
   }
+
+  // TODO: Revisit configuration and persistence of configuration.
+  // https://geomesa.atlassian.net/browse/GEOMESA-958
+  // https://geomesa.atlassian.net/browse/GEOMESA-984
+  post("/ds/?") {
+    logger.debug("In ds registration method")
+
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+
+    if (ds == null) {
+      NotFound(reason = "Could not load data store using the provided parameters.")
+    } else {
+      this.synchronized {
+        abs = new AccumuloBlobStore(ds)
+      }
+      //see https://httpstatuses.com/204
+      NoContent()
+    }
+  }
+
+  // scalatra routes bottom up, so we want the ds post to be checked first
+  post("/") {
+    try {
+      logger.debug("In file upload post method")
+      if (abs == null) {
+        NotFound(reason = "AccumuloBlobStore is not initialized.")
+      } else {
+        fileParams.get("file") match {
+          case None =>
+            halt(400, reason = "no file parameter in request")
+          case Some(file) =>
+            val otherParams = multiParams.toMap.map { case (s, p) => s -> p.head }
+            val tempFile = new File(tempDir.toString + '/' + file.getName)
```

Comment:
Here's a brief overview of some of the security risks:
http://security.stackexchange.com/questions/503/what-are-the-security-risks-of-letting-the-users-upload-content-to-my-site
In addition to that, if you e.g. knew that someone was uploading a file named 'myreport', you could maliciously replace that file by uploading something with the same name. You have to remember that web services are multi-threaded and serve concurrent users. Thus, ensuring users can't impact each other is important.